### ISSUE TITLE
Screen reader user cannot tell that Electron desktop now has basic screen reader support

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#12321]
### System details

    RStudio Edition : Desktop
    RStudio Version : Elsbeth Geranium daily builds
    OS Version      : Any
    R Version       : Any

### Steps to reproduce the problem

Start a screen reader, then start RStudio Desktop (Electron). Try to do anything (as a blind user).

### Describe the problem in detail

Everything in the RStudio window is invisible to the screen reader software. All it can tell you is that there's a window. This is because we force Chromium accessibility support off by default. This is a carryover from an issue we had seen years ago with the Magnet application (on macOS) triggering Chromium's accessibility support and causing performance issues with Qt. #1990

The code that does this was ported over from Qt as-is (by me, for full disclosure).

https://github.com/rstudio/rstudio/blob/f8461c63eb68ea77943d488ee6c075de3977cd05/src/node/desktop/src/main/main.ts#L86-L93

As a blind user, you would need to know to go to the main menu, then Help / Accessibility / Screen Reader Support and restart the application before you could actually do anything via screen reader. That is unacceptable, and although the IDE is still far from completely usable via screen reader, this masks all the significant improvements we've made by the switch to Electron.

### Describe the behavior you expected

We should not disable Chromium's automatic accessibility support by default and let Chromium do the right thing.

In my own testing, the Magnet issue is no longer a problem; don't know if that's due to improvements in Magnet, or Chromium (via Electron), or macOS.

That said, it's possible there could be other such corner cases where a user would want to disable all Chromium accessibility (I vaguely recall there was a problem like that on Linux, too, at one point), so we should provide some mechanism to do this, but it shouldn't be the default!

### Proposed Fix

There are two aspects of screen reader support in RStudio Electron.

1. Chromium accessibility support, which enables the APIs used by accessibility tools such as screen reader software. By default, Chromium (including Electron and the Chrome Web Browser) will enable this support automatically if it detects a screen reader is in use. This is what we're currently disabling.
2. Additional RStudio-specific behaviours controlled by a user setting (Global Options / Accessibility / Screen Reader Support). This turns on additional RStudio-specific things such telling the terminal pane (xterm) to provide a DOM equivalent to usual non-DOM-based rendering (which is invisible to screen readers).

Right now, in Electron, the "screen reader support enable/disable" user preference controls both of those. It needs to only control #2. That would make it match the behaviour of RStudio Server/Workbench.

We could add another setting for #1 ("Disable Chromium Accessibility").

This would be a simple and low risk change (that I'm happy to make and provide detailed instructions on how to test). I'd urge us to take this change for Elsbeth Geranium, or our story about improved accessibility via Electron is severely hamstrung.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by mikebessuille** [comment]
We agree- this is very important for Elsbeth.   I'm not sure I'd delay Elsbeth for this, but if we don't get it, we should release a patch with this support.

**Comment by gtritchie** [comment]
That's because without any accessibility tree coming from Chromium, NVDA is flying blind (no pun intended) and is defaulting to browse mode, where it uses single-key commands.

You can switch out of browse mode, into focus mode, using NVDA+Spacebar (where "NVDA" is usually the Insert key, but could also be configured as the Capslock key).

**Comment by gtritchie** [comment]
More on focus vs browse: https://www.accessibility-developer-guide.com/knowledge/screen-readers/desktop/browse-focus-modes/

Note that VoiceOver (macOS) doesn't normally make such a distinction, but you can switch VoiceOver into a somewhat similar "browse" mode via VO+Q. VoiceOver calls it "single key quick-nav".

**Comment by gtritchie** [comment]
JAWS does the same (starts in browse mode when Chromium accessibility is disabled).

Narrator is the exception; it starts in regular mode and has to be switched into (what Narrator calls) scan mode.

**Comment by ronblum** [comment]
Thanks! Confirmed—it works as expected then (even though it's not a very useful scenario).

Note: In NVDA, <kbd>Insert-Space</kbd> switches to focus mode.
